src/core/scenario.jl

```diff
@@ -367,6 +377,106 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
+        if haskey(d["GHP"], "ghpghx_responses") && !isempty(d["GHP"]["ghpghx_responses"])
+            get_ghpghx_from_input = true
+        end        
+    elseif haskey(d, "GHP") && !haskey(d["GHP"],"building_sqft")
+        @error("If evaluating GHP you must enter a building_sqft")
+    end            
+    # Modify Heating and Cooling loads for GHP retrofit to account for HVAC VAV efficiency gains
+    if eval_ghp
+        # Assign efficiency_thermal_factors if not specified (and if applicable to building type and climate zone)
+        for factor in [("space_heating_efficiency_thermal_factor", "heating"), ("cooling_efficiency_thermal_factor", "cooling")]
+            if isnan(d["GHP"][factor[1]])
+                assign_thermal_factor!(d, factor[2])
+            end
+        end
+        heating_thermal_load_reduction_with_ghp_kw = space_heating_load.loads_kw * (1.0 - d["GHP"]["space_heating_efficiency_thermal_factor"])
+        cooling_thermal_load_reduction_with_ghp_kw = cooling_load.loads_kw_thermal * (1.0 - d["GHP"]["cooling_efficiency_thermal_factor"])
+    end
+    # Call GhpGhx.jl module if only ghpghx_inputs is given, otherwise use ghpghx_responses
+    if eval_ghp && !(get_ghpghx_from_input)
+        if d["GHP"]["ghpghx_inputs"] in [nothing, []]
+            number_of_ghpghx = 1
+            d["GHP"]["ghpghx_inputs"] = [Dict()]
+        else
+            number_of_ghpghx = length(d["GHP"]["ghpghx_inputs"])
+        end
+        # Call PVWatts for hourly dry-bulb outdoor air temperature
+        ambient_temperature_f = []
+        if !haskey(d["GHP"]["ghpghx_inputs"][1], "ambient_temperature_f") || isempty(d["GHP"]["ghpghx_inputs"][1]["ambient_temperature_f"])
+            url = string("https://developer.nrel.gov/api/pvwatts/v6.json", "?api_key=", nrel_developer_key,
+                    "&lat=", d["Site"]["latitude"] , "&lon=", d["Site"]["longitude"], "&tilt=", d["Site"]["latitude"],
+                    "&system_capacity=1", "&azimuth=", 180, "&module_type=", 0,
+                    "&array_type=", 0, "&losses=", 0.14, "&dc_ac_ratio=", 1.1,
+                    "&gcr=", 0.4, "&inv_eff=", 99, "&timeframe=", "hourly", "&dataset=nsrdb",
+                    "&radius=", 100)
+            try
+                @info "Querying PVWatts for ambient temperature"
+                r = HTTP.get(url)
+                response = JSON.parse(String(r.body))
+                if r.status != 200
+                    error("Bad response from PVWatts: $(response["errors"])")
+                end
+                @info "PVWatts success."
+                temp_c = get(response["outputs"], "tamb", [])
+                if length(temp_c) != 8760 || isempty(temp_c)
+                    @error "PVWatts did not return a valid temperature profile. Got $temp_c"
+                end
+                ambient_temperature_f = temp_c * 1.8 .+ 32.0
+            catch e
+                @error "Error occurred when calling PVWatts: $e"
+            end
```

Comment:
Yes, I agree we should probably make a function for calling PVWatts and pass an argument whether we want PV prodfactor or ambient temperature or both depending on if PV and/or GHP is considered. Since the PV PVWatts call is currently embedded in the prodfactor(PV) function which gets called in reopt_inputs, I'm going to punt on this for now to avoid a bit of work rearranging that and the conditionals for when GHP needs the PVWatts call too.  It won't effect the user interaction, so we can make the efficiency improvement later without any major version change.

Reply:
Added https://github.com/NREL/REopt.jl/issues/113 to address this later